### PR TITLE
Use async requests and temporary files for signing

### PR DESCRIPTION
**Pull request by catlee**
Also: drop py3.6, add 3.8 for signingscript

### CODE REVIEW

File: signingscript/requirements/base.py36.txt [hunk 1]
```diff
@@ -1,272 +0,0 @@
-# SHA1:e2e6bd7fcef883d3413ffa98e29a0ac85688d400
```
Comment (escapewindow) [inline signingscript/requirements/base.py36.txt:1]: Should we create `{base,test}.py38.txt` files? They may diverge from py37 at some point. I suppose it depends on what's using them.

File: signingscript/setup.py [hunk 2]
```diff
@@ -4,10 +4,16 @@
 with open(os.path.join(os.path.abspath(os.path.dirname(__file__)), "version.txt")) as f:
     version = f.read().rstrip()
 
-with open(
-    os.path.join(os.path.abspath(os.path.dirname(__file__)), "requirements/base.in")
-) as f:
-    install_requires = f.readlines()
+install_requires = [
```
Comment (escapewindow) [inline signingscript/setup.py:7]: Hm, did I see something's broken if we don't do this?
If so, what's broken? Do we need to package `requirements/base.in` ?

File: signingscript/src/signingscript/sign.py [hunk 3]
```diff
@@ -11,19 +11,19 @@
...
-# TODO: Use aiohttp for this.
-import requests
 import shutil
 import subprocess
 import sys
 import tarfile
 import tempfile
 import zipfile
 
+from io import BytesIO
```
Comment (escapewindow) [inline signingscript/src/signingscript/sign.py:23]: Did you figure out what was hanging? New tempfile per retry?

**Review by escapewindow — APPROVED** [review]

File: signingscript/requirements/base.py36.txt (hunk 1 above)
Comment (catlee) [reply signingscript/requirements/base.py36.txt:1]: not until we need it

File: signingscript/setup.py (hunk 2 above)
Comment (catlee) [reply signingscript/setup.py:7]: This breaks if you want to add URLs into `requirements/base.in`, e.g. to a URL on github for testing a prerelease version of one of the dependencies.

File: signingscript/src/signingscript/sign.py (hunk 3 above)
Comment (catlee) [reply signingscript/src/signingscript/sign.py:23]: threads deadlocking. I was using a thread to call winsign.sign.sign_file, and aiohttp also uses threads for reading from the temporary files. With so many concurrent requests, the threadpool wasn't large enough, and so deadlocked.

The solution was to make sign_file async so I don't need the first level of threads.

**Pull request merged**